indexer/workers/importer.py

```diff
@@ -154,14 +153,23 @@ def index_routing(self, publication_date_str: Optional[str]) -> str:
         if publication_date_str:
             try:
                 year = datetime.strptime(publication_date_str, "%Y-%m-%d").year
+                current_year = datetime.now().year
+
+                if year > current_year:
+                    year = -1
+                    logger.warning(
+                        f"Publication date greater than current year: {current_year}"
+                    )
             except ValueError as e:
-                logger.error(f"Error parsing date: {str(e)}")
+                logger.warning(f"Error parsing date: {str(e)}")
 
         index_name_prefix = os.environ.get("ELASTICSEARCH_INDEX_NAME_PREFIX")
-        if year and year >= 2021:
+        if 2021 <= year <= current_year:
```

Comment:
```suggestion
        if year >= 2021:
```
Since you've already handled the `if year > current_year` part.